tests/dataframe/test_shuffles.py

```diff
@@ -40,7 +40,7 @@ def generator(
 def pre_shuffle_merge_ctx():
     """Fixture that provides a context manager for pre-shuffle merge testing."""
 
-    def _ctx(threshold: int | None = None):
+    def _ctx(threshold: Optional[int] = None):
```

Comment:
for some reason this is now failing on python 3.9 unit tests. might be because we didn't actually set the version to 3.9 in the past? 🤷